core/src/main/kotlin/io/customer/sdk/communication/EventBus.kt

```diff
@@ -55,4 +61,17 @@ class EventBusImpl(
         jobs.forEach { it.cancel() }
         jobs.clear()
     }
+
+    override fun <T : Event> subscribe(type: KClass<T>, action: suspend (T) -> Unit): Job {
+        val job = scope.launch {
+            flow.filter { type.isInstance(it) }.mapNotNull { it.safeCast(type) }.collect { event ->
+                action(event)
+            }
+        }
+        jobs.add(job)
+        return job
+    }
+    private fun <T : Event> Event.safeCast(type: KClass<T>): T? {
+        return if (type.isInstance(this)) type.cast(this) else null
```

Comment:
Can't we use Kotlin's predefined safeCast here like this?

```suggestion
        return type.safeCast(this)
```

Or may be remove this method and just use Kotlin one 🤔 

```kotlin
IdentifyEvent::class.safeCast(event)
```

Reply:
I used it first but then I saw this
```
@WasExperimental(ExperimentalStdlibApi::class)
```
and thought it said experimental but it said `was` :D